Надо подкачаться

sorrowfullight opened the pull request:
1. добавлен механизм отправки и получения данных с сервера;
2. удален файл data.js с моковыми данными;
3. добавлены окна успешной/ неуспешной загрузки изображения;
4. изменен файл-точка входа main.js;

---
:mortar_board: [Надо подкачаться](https://up.htmlacademy.ru/javascript-individual/1/user/2400143/tasks/17)

:boom: https://htmlacademy-javascript.github.io/2400143-kekstagram-1/10/

sorrowfullight committed:
1. добавлен механизм отправки и получения данных с сервера;
2. удален файл data.js с моковыми данными;
3. добавлены окна успешной/ неуспешной загрузки изображения;
4. изменен файл-точка входа main.js;

saimon395 commented on js/form.js:
```diff
@@ -87,12 +98,31 @@ pristine.addValidator(
+    } finally {
+      unblockSubmitButton();
+      closeUpload();
+    }
```
перенеси closeUpload в try, при ошибке форма закрываться не должна, и сообщение об ошибке должно поверх показываться

saimon395 commented on js/form.js:
```diff
@@ -87,12 +98,31 @@ pristine.addValidator(
+  //     showAlert(err.message);
+  //   }
+  // )
+
```
закоментированный код удаляй

saimon395 commented on js/message.js:
```diff
@@ -0,0 +1,57 @@
+    successMessage.remove();
+    document.body.classList.remove('modal-open');
+  };
+  successButton.addEventListener('click', (evt) => {
+    evt.preventDefault();
+    hideSuccessMessage();
+  });
+  document.addEventListener('keydown', () => {
+    if (isEscapeKey) {
+      hideSuccessMessage();
+    }
+  });
+  document.addEventListener('click', (evt) => {
+    if (!evt.target.closest('.success__inner') && !evt.target.closest('.success__button')) {
+      hideSuccessMessage();
+    }
+  });
```
сейчас есть проблема что обработчик не удаляются, тебе в hideSuccessMessage нужно удалять обработчики и на click и на keydown

saimon395 commented on js/util.js:
```diff
@@ -36,4 +38,25 @@ const createIdGenerator = () => {
+  }, ALERT_SHOW_TIME);
+};
+
+export {getRandomInteger,getRandomArrayElement,createRandomIdFromRangeGenerator,createIdGenerator,isEscapeKey,showAlert};
```
неиспользуемый функции типа getRandomInteger можешь уже удалять

sorrowfullight committed:
1. исправлены недочеты, в том числе висящие обработчики

saimon395 commented on js/api.js:
```diff
@@ -1,7 +1,7 @@
   GET_DATA: '/data',
-  SEND_DATA: '/',
+  SEND_DATA: '/jjj',
 };
```
вот это поправить не забудь )

saimon395 commented on js/form.js:
```diff
@@ -109,20 +108,14 @@ imageForm.addEventListener('submit', async (evt) => {
       unblockSubmitButton();
-      closeUpload();
+      document.removeEventListener('keydown', onDocumentKeydown);
     }
```
не надо здесь удалять обработчик, иначе при ошибке запроса, не получается закрыть форму через esc, удаляй обработчик в функции закрытия формы

sorrowfullight committed:
1. исправлены недочеты в api.js и form.js

sorrowfullight committed:
1. исправлены недочеты
2. в файле form.js введена переменная onCloseUpload для удаления обработчика onDocumentKeydown (иначе линтор выдает ошибку при удалении обработчика внутри функции closeUpload)

saimon395 commented on js/message.js:
```diff
@@ -0,0 +1,58 @@
+      hideErrorMessage();
+    });
+  };
+  const onModalClose = (evt) => {
+    if (errorButtonHandler || isEscapeKey(evt) || !evt.target.closest('.error__inner') && !evt.target.closest('.error__button')) {
+      hideErrorMessage();
+      document.removeEventListener('click', onModalClose);
+      document.removeEventListener('keydown', onModalClose);
+    }
+  };
+  document.addEventListener('keydown', onModalClose);
+  document.addEventListener('click', onModalClose);
+};
```
теперь проблема другая, при клике на esc закрывается и форма и сообщение об ошибке , а по тз при закрытии этого сообщения форма должна оставаться открытой https://up.htmlacademy.ru/javascript-individual/1/project/kekstagram#kekstagram-3-5

saimon395 replied:
ну и обработчики ещё раз проверь, сейчас если открыт форму и сразу её закрыть, нажав на крестик, остаётся висеть обработчик на keydown

sorrowfullight committed:
1. исправлены недочеты с обработчиками
